webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/reports/ReportingServiceImpl.scala

```diff
@@ -455,7 +434,7 @@ trait DefaultFindRuleNodeStatusReports extends ReportingService {
     } yield {
       val t2 = System.currentTimeMillis
       //we want to have nodeStatus for all asked node, not only the ones with reports
-      val nodeStatusReports = runInfos.map { case (nodeId, runInfo) =>
+      val nodeStatusReports = runInfos.par.map { case (nodeId, runInfo) =>
```

Comment:
this part hasn't been tested yet

Reply:
but without the par, with 5000 nodes it takes 22s

Reply:
this is a linear improvement

Reply:
note for reference: this causes deadlocks - see https://issues.rudder.io/issues/16256